src/app/post-list/post-list.component.ts

```diff
@@ -0,0 +1,45 @@
+import {Component, OnInit} from '@angular/core';
+import {PostService} from "../services/post.service";
+
+@Component({
+    selector: 'app-post-list',
+    templateUrl: './post-list.component.html',
+    styleUrls: ['./post-list.component.scss']
+})
+export class PostListComponent implements OnInit {
+    posts: any[] = [];
+
+    constructor(private postService: PostService) {
+    }
+
+    ngOnInit(): void {
+        this.postService.getPosts().subscribe((data: any) => {
+            this.posts = data;
+        });
+    }
+
+  formatPublicationDate(dateString: string): string {
```

Comment:
This better works implemented as a pipe. If you supply a function, like this, Angular has no choice but to consider it impure and evaluate it on every render, even if the parameter `dateString` is the same and wouldn't normally require a render.

You can try to implement such a pipe yourself, or just add something like [time-ago-pipe - npm](https://www.npmjs.com/package/time-ago-pipe). This "time ago" feature is worth its own package.